homeassistant/helpers/entity_component.py

```diff
@@ -348,14 +349,33 @@ def async_reset(self):
             self._async_unsub_polling()
             self._async_unsub_polling = None
 
-    @callback
+    @asyncio.coroutine
     def _update_entity_states(self, now):
         """Update the states of all the polling entities.
 
         This method must be run in the event loop.
         """
-        for entity in self.platform_entities:
-            if entity.should_poll:
-                self.component.hass.async_add_job(
-                    entity.async_update_ha_state(True)
-                )
+        # protect core for flooting all executors
+        if self._process_updates:
+            return
+        self._process_updates = True
+
+        try:
+            tasks = []
+            for entity in self.platform_entities:
+                if entity.should_poll:
+                    task = self.component.hass.async_add_job(
```

Comment:
We should never return the task from async_add_job. You add a job because you don't care about the result. If you care about the result you yield from it.